.gitlab/benchmarks/macrobenchmarks.yml

```diff
@@ -0,0 +1,353 @@
+stages:
+  - check-azure-pipeline
+  - benchmarks
+
+variables:
+  MACROBENCHMARKS_CI_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/benchmarking-platform:dotnet-throughput-2
+
+check_azure_pipeline:
+  stage: check-azure-pipeline
+  image: $MACROBENCHMARKS_CI_IMAGE
+  script:
+    - git clone --branch dd-trace-dotnet/macro https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.ddbuild.io/DataDog/benchmarking-platform platform && cd platform
+    - ./wait-for-pipeline.sh
+  artifacts:
+    name: "artifacts"
+    when: always
+    paths:
+      - artifacts/
+      - build-id.txt
+    expire_in: 3 months
+  interruptible: true
+  tags: ["arch:amd64"]
+  timeout: 1h
+  rules:
+    - when: on_success
+
+.benchmarks:
+  stage: benchmarks
+  tags: ["runner:apm-k8s-same-cpu"]
+  needs: ["check_azure_pipeline"]
+  interruptible: true
+  timeout: 1h
+  retry:
+    max: 2
+    when:
+      - unknown_failure
+      - data_integrity_failure
+      - runner_system_failure
+      - scheduler_failure
+      - api_failure
+  rules:
+    - when: on_success
+  image: $MACROBENCHMARKS_CI_IMAGE
+
+  script:
+    - source build-id.txt
+    - echo "Building for the following build https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=$buildId&view=results"
+    - git clone --branch dd-trace-dotnet/macro https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.ddbuild.io/DataDog/benchmarking-platform platform && cd platform
+    - bp-runner bp-runner.yml --debug
+  artifacts:
+    name: "artifacts"
+    when: always
+    paths:
+      - artifacts/
+    expire_in: 3 months
+  variables:
+    FF_USE_LEGACY_KUBERNETES_EXECUTION_STRATEGY: "true"
+
+    K6_OPTIONS_WARMUP_RATE: 30
```

Comment:
Just a comment, K6_OPTIONS_WARMUP_RATE seems to be too low, I would suggest raising to between K6_OPTIONS_NORMAL_OPERATION_RATE and K6_OPTIONS_HIGH_LOAD_RATE (400 to 600).

We will need to tweak RPS after there is some data in dashboards, so no action is needed right now.